r-package/R/utils.R

```diff
@@ -102,73 +102,75 @@ download_gpkg <- function(file_url, progress_bar = showProgress){
 
   if( !(progress_bar %in% c(T, F)) ){ stop("Value to argument 'showProgress' has to be either TRUE or FALSE") }
 
-## one single file
-
+  ## one single file
   if(length(file_url)==1 & progress_bar == TRUE){
-
     # download file
     temps <- paste0(tempdir(),"/", unlist(lapply(strsplit(file_url,"/"),tail,n=1L)))
-    httr::GET(url=file_url, httr::progress(), httr::write_disk(temps, overwrite = T))
+    
+    if (!file.exists(temps)) {
+      httr::GET(url=file_url, httr::progress(), httr::write_disk(temps, overwrite = T))
+    }
 
     # load gpkg
     temp_sf <- load_gpkg(file_url, temps)
     return(temp_sf)
-
-
-    }
+  }
 
   else if(length(file_url)==1 & progress_bar == FALSE){
-
     # download file
     temps <- paste0(tempdir(),"/", unlist(lapply(strsplit(file_url,"/"),tail,n=1L)))
```

Comment:
in order to account for files with similar names, this line of code should be replaced by

```
    temps <- paste0(tempdir(),"\\", sub(".*ipea.gov.br/", "", file_url) )
    temps <- gsub("/", "-", temps, fixed=TRUE)
```